lib/src/main/java/io/ably/lib/util/AblyAgentValidator.java

```diff
@@ -0,0 +1,45 @@
+package io.ably.lib.util;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class AblyAgentValidator {
+    /**
+     * Agent name validation regex.
+     * Allow only lowercase letters and '-'.
```

Comment:
I looked at some examples and the javascript PR and it looked like the names only have lowercase letters and dashes :wink: Not sure if that's a correct assumption though :thinking:

Reply:
On this point I've just worked up a new docs PR to address this question, in response to https://github.com/ably/ably-cocoa/pull/1119 also. See https://github.com/ably/docs/pull/1096.

Reply:
I've changed the regex https://github.com/ably/ably-java/pull/671/commits/3fb4f4978826d5db3dcc1e1176e5c5b63a582202 but still, I wasn't sure if uppercase letters are allowed so I've stuck to lowercase letters only, is that correct? :thinking: